### PR TITLE
Add some lit v2 support and tidy up useless-literals rule

### PR DESCRIPTION
**Pull request by 43081j**
Still need to get the coverage back up because i added a bunch of complexity to compute property maps.

### CODE REVIEW

**Comment by 43081j** [comment]
I suppose we need to understand things like:

`<foo ${directive()}>` - an element expression, which presumably means the directive is given an `ElementPart`

`<${tag}>` - an element name from using `literal` is now possible

**Comment by stramel** [comment]
Will try to throw in a review this week

**Comment by 43081j** [comment]
FYI the changes are fairly big for the useless literals rule because it has changed from:

"any literal in any template"

to:

"any literal in any template which isn't a property binding"

which now means it needs knowledge of the HTML AST... so it uses the analyzer

File: src/rules/no-useless-template-literals.ts [hunk 1]
```diff
@@ -23,26 +24,83 @@ const rule: Rule.RuleModule = {
...
+            const expr = node.quasi.expressions[i];
+            if (
+              expr.type === 'Literal' &&
+              !node.quasi.quasis[i].value.raw.endsWith('=')
+            ) {
               context.report({
                 node: expr,
-                message: 'Literals must not be substituted into templates'
+                message:
+                  'Literals must not be substituted into ' + 'text bindings'
```
Comment (43081j) [inline src/rules/no-useless-template-literals.ts:75]: formatter did it better, can remove the ` + `

**Comment by stramel** [comment]
@43081j Shoot, sorry about not getting to this one. I thought the other one was what I had committed to reviewing. Will get on this ASAP.

**Review by stramel — APPROVED** [review]

**Pull request merged**